src/lib/socialimagesdatabase.cpp

```diff
@@ -0,0 +1,520 @@
+        QList<SocialImage::ConstPtr> images;
+    } result;
+
+    QSqlQuery m_singleImageQuery;
+};
+
+SocialImagesDatabasePrivate::SocialImagesDatabasePrivate(SocialImagesDatabase *q)
+    : AbstractSocialCacheDatabasePrivate(
+            q,
+            SocialSyncInterface::socialNetwork(SocialSyncInterface::Facebook),
+            SocialSyncInterface::dataType(SocialSyncInterface::Images),
+            QLatin1String(DB_NAME),
+            VERSION)
+{
+}
+
+SocialImagesDatabasePrivate::~SocialImagesDatabasePrivate()
+{
+}
+
+void SocialImagesDatabasePrivate::clearCachedImages(QSqlQuery &query)
+{
+    while (query.next()) {
+        QString image = query.value(0).toString();
+        if (!image.isEmpty()) {
+            QFile imageFile(image);
+            if (imageFile.exists()) {
+                imageFile.remove();
+            }
+        }
+    }
+}
+
+QList<SocialImage::ConstPtr> SocialImagesDatabasePrivate::queryImages(int accountId,
+                                                                      const QDateTime &olderThan)
+{
+    Q_Q(SocialImagesDatabase);
+
+    QList<SocialImage::ConstPtr> data;
+
+    QString queryString = QLatin1String("SELECT accountId, "
+                                        "imageUrl, imageFile, createdTime, expires "
+                                        "FROM images "
+                                        "WHERE accountId = :accountId");
+
+    if (olderThan.isValid()) {
+        queryString.append(" AND createdTime < :createdTime");
+    }
+
+    QSqlQuery query = q->prepare(queryString);
+    query.bindValue(":accountId", accountId);
+
+    if (olderThan.isValid()) {
+        query.bindValue(":createdTime", olderThan.toTime_t());
+    }
+
+    if (!query.exec()) {
+        qWarning() << Q_FUNC_INFO << "Failed to query images:" << query.lastError().text();
+        return data;
+    }
+
+    while (query.next()) {
+        data.append(SocialImage::create(query.value(0).toInt(),                             // accountId
+                                        query.value(1).toString(),                          // imageUrl
+                                        query.value(2).toString(),                          // imageFile
+                                        QDateTime::fromTime_t(query.value(3).toUInt()),     // createdTime
+                                        QDateTime::fromTime_t(query.value(4).toUInt())));   // expires
+    }
+
+    return data;
+}
+
+QList<SocialImage::ConstPtr> SocialImagesDatabasePrivate::queryExpired(int accountId)
+{
+    Q_Q(SocialImagesDatabase);
+
+    QList<SocialImage::ConstPtr> data;
+
+    int currentTime = QDateTime::currentDateTime().toTime_t();
+
+    QString queryString = QLatin1String("SELECT accountId, "
+                                        "imageUrl, imageFile, createdTime, expires "
+                                        "FROM images "
+                                        "WHERE accountId = :accountId AND expires < :currentTime");
```

Comment:
if the user changes their datetime or switches timezone, could that trigger a premature deletion?  I guess it's not an issue, since in that case we'd just redownload the image and it would then get a new expiry time in the new timezone etc...